en/lessons/advanced/nerves.md

```diff
@@ -0,0 +1,263 @@
+
+After configuring the LEDs itself, we surely need to control them somehow. To do that, we will add a GenServer in `lib/network_led/blinker.ex` with these contents:
+
+```elixir
+defmodule NetworkLed.Blinker do
+  use GenServer
+
+  @moduledoc """
+    Simple GenServer to control GPIO #18.
+  """
+
+  require Logger
+  alias Nerves.Leds
+
+  def start_link(state \\ []) do
+    GenServer.start_link(__MODULE__, state, name: __MODULE__)
+  end
+
+  def init(state) do
+    enable()
+
+    {:ok, state}
+  end
+
+  def handle_cast(:enable, state) do
+    Logger.info("Enabling LED")
+    Leds.set(green: true)
+
+    {:noreply, state}
+  end
+
+  def handle_cast(:disable, state) do
+    Logger.info("Disabling LED")
+    Leds.set(green: false)
+
+    {:noreply, state}
+  end
+
+  def enable() do
+    GenServer.cast(__MODULE__, :enable)
+  end
+
+  def disable() do
+    GenServer.cast(__MODULE__, :disable)
+  end
+end
+
+```
+
+And to enable it you need to also add it to supervision tree in `lib/network_led/application.ex`: add `{NetworkLed.Blinker, name: NetworkLed.Blinker}` under the `  def children(_target) do` group. 
+
+Notice that Nerves has two different supervision trees in application - one for the host machine and one for actual devices. 
+
+After this - that's it! You actually can upload the firmware and check that `NetworkLed.Blinker.disable()` disables LED (which is enabled by default in code), and `NetworkLed.Blinker.enable()` enables it.
+
+We have control over the LED from the command prompt! 
+
+Now the only missing piece of the puzzle we need is controlling it via the web interface.
+
+## Adding the web server
+
+In this step, we will be using `Plug.Router`. If you need a reminder - feel free to skim through the [Plug](../../../lessons/specifics/plug/) lesson.
+
+First, we will add `{:plug_cowboy, "~> 2.0"},` to `mix.exs` and install the dependencies. 
+
+Then, add the actual process to process those requests in `lib/network_led/http.ex` :
+
+```elixir
+defmodule NetworkLed.Http do
+  use Plug.Router
+
+  plug(:match)
+  plug(:dispatch)
+
+  get("/", do: send_resp(conn, 200, "Feel free to use API endpoints!"))
+
+  get "/enable" do
+    NetworkLed.Blinker.enable()
+    send_resp(conn, 200, "LED enabled")
+  end
+
+  get "/disable" do
+    NetworkLed.Blinker.disable()
+    send_resp(conn, 200, "LED disabled")
+  end
+
+  match(_, do: send_resp(conn, 404, "Oops!"))
+end
+```
+
+And, the final step - add `{Plug.Cowboy, scheme: :http, plug: NetworkLed.Http, options: [port: 80]}` to the application supervision tree.
+
+After the firmware update, you can try it! `http://192.168.88.2/` is returning plain text response, and `http://192.168.88.2/enable` with `http://192.168.88.2/disable` disable and enable that LED!
+
+You can even pack Phoenix-powered user interfaces into your Nerves app, however, it [will require some tweaking](https://github.com/nerves-project/nerves/blob/master/docs/User%20Interfaces.md#phoenix-web-interfaces). 
+
```

Comment:
Thanks @brain-geek!

All in all very quick and informative, unfortunately I cannot test it in my Rpi3 as my sd card reader has left for a better place. I am going to buy a new one though and check again, but it seems solid as a starting lesson.

I have left some minor corrections for you.

Reply:
Thanks a lot @gemantzu , applied all your suggestions!